pkg/kubelet/metrics/collectors/emptydir_metrics.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collectors
+
+import (
+	"context"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/component-base/metrics"
+	"k8s.io/klog/v2"
+
+	kubeletmetrics "k8s.io/kubernetes/pkg/kubelet/metrics"
+	serverstats "k8s.io/kubernetes/pkg/kubelet/server/stats"
+)
+
+var (
+	emptyDirUsedBytesDesc = metrics.NewDesc(
+		metrics.BuildFQName(
+			"",
+			kubeletmetrics.KubeletSubsystem,
+			kubeletmetrics.EmptyDirUsedBytesKey,
+		),
+		"Bytes used by the emptyDir volume.",
+		[]string{
+			"volume_name",
+			"namespace",
+			"pod",
+		},
+		nil,
+		metrics.ALPHA,
+		"",
+	)
+)
+
+type emptyDirMetricsCollector struct {
+	metrics.BaseStableCollector
+
+	statsProvider serverstats.Provider
+}
+
+// Check if emptyDirMetricsCollector implements necessary interface
+var _ metrics.StableCollector = &emptyDirMetricsCollector{}
+
+// NewEmptyDirMetricsCollector implements the metrics.StableCollector interface and
+// exposes metrics about pod's emptyDir.
+func NewEmptyDirMetricsCollector(statsProvider serverstats.Provider) metrics.StableCollector {
```

Comment:
maybe check why we return `metrics.StableCollector` instead of `emptyDirMetricsCollector`